### PR TITLE
Correctly report status code and operation ID on S3 gateway

### PR DESCRIPTION
**Pull request by arielshaqed**
Previously we incorrectly reported only status 200.  Also some unknown
operations are misreported as "", meaning Grafana makes up a long
incomprehensible list of labels for them.

### CODE REVIEW

File: pkg/gateway/handler.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func NewHandler(region string, catalog catalog.Interface, multipartsTracker mult
 func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	setDefaultContentType(w, req)
 	o := req.Context().Value(ContextKeyOperation).(*operations.Operation)
-	operationHandler := h.operationHandlers[o.OperationID]
+	operationHandler := h.operationHandlers[o.FooOperationID]
```
Comment (nopcoder) [inline pkg/gateway/handler.go:129]: can we revert this change?

File: pkg/gateway/middleware.go [hunk 2]
```diff
@@ -127,7 +127,8 @@ func DurationHandler(next http.Handler) http.Handler {
 		o := ctx.Value(ContextKeyOperation).(*operations.Operation)
 		start := time.Now()
 		mrw := httputil.NewMetricResponseWriter(w)
-		next.ServeHTTP(w, req)
+		next.ServeHTTP(mrw, req)
+		operationID := o.OperationID
```
Comment (nopcoder) [inline pkg/gateway/middleware.go:131]: can revert this one too

File: pkg/gateway/middleware.go [hunk 3]
```diff
@@ -175,7 +176,7 @@ func OperationLookupHandler(next http.Handler) http.Handler {
 		ctx := req.Context()
 		o := ctx.Value(ContextKeyOperation).(*operations.Operation)
 		repoID := ctx.Value(ContextKeyRepositoryID).(string)
-		var operationID operations.OperationID
+		operationID := operations.OperationIDOperationNotFound
```
Comment (nopcoder) [inline pkg/gateway/middleware.go:179]: I think we need to update the structure (o.OperationID) at this point as we do return in case of an error.

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/gateway/handler.go (hunk 1 above)
Comment (arielshaqed) [reply pkg/gateway/handler.go:129]: YIKES!

File: pkg/gateway/middleware.go (hunk 3 above)
Comment (arielshaqed) [reply pkg/gateway/middleware.go]: removed `operationID` entirely :-)

**Review by arielshaqed — COMMENTED** [review]
Thanks!

PTAL...

File: pkg/gateway/middleware.go (hunk 2 above)
Comment (nopcoder) [reply pkg/gateway/middleware.go:131]: Still we don't use this one, based on diff

**Review by nopcoder — APPROVED** [review]
Just one unused local var I think we can remove, but looks good - thanks!
